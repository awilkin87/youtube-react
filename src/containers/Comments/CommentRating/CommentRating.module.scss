@use '../../../styles/shared.module';

.container {
  display: inline-flex;
  font-size: 13px;
  color: shared.$secondary-text-color;

  // Override Semantic UI color
  :global(i.icon) {
    color: #909090;
  }
  
  .thumbsUp {
    margin-right: 14px;
  }

  .thumbsDown {
    margin-right: 14px;
  }
}
