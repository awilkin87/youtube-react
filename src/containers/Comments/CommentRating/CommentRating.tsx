import { Icon } from 'semantic-ui-react';
import styles from './CommentRating.module.scss';

export interface CommentRatingProps {
  likes: number;
}

const CommentRating: React.FunctionComponent<Readonly<CommentRatingProps>> = function ({ likes }) {
  return (
    <div className={styles.container}>
      <div className={styles.thumbsUp}>
        <Icon name='thumbs up' />
        <span>{likes || null}</span>
      </div>
      <div className={styles.thumbsDown}>
        <Icon name='thumbs down' />
      </div>
    </div>
  );
}

export default CommentRating;
