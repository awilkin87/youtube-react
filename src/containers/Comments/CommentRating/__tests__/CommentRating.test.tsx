import { shallow } from "enzyme";
import CommentRating from '../CommentRating';

describe('CommentRating', () => {
  test('Renders CommentRating with non-zero likes', () => {
    const wrapper = shallow(<CommentRating likes={2} />);
    expect(wrapper).toMatchSnapshot();
  });

  
  test('Renders CommentHeader with zero likes', () => {
    const wrapper = shallow(<CommentRating likes={0} />);
    expect(wrapper).toMatchSnapshot();
  });
});
