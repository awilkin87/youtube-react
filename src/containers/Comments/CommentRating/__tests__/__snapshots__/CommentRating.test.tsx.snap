// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentRating Renders CommentHeader with zero likes 1`] = `
<div
  className="container"
>
  <div
    className="thumbsUp"
  >
    <Icon
      as="i"
      name="thumbs up"
    />
    <span />
  </div>
  <div
    className="thumbsDown"
  >
    <Icon
      as="i"
      name="thumbs down"
    />
  </div>
</div>
`;

exports[`CommentRating Renders CommentRating with non-zero likes 1`] = `
<div
  className="container"
>
  <div
    className="thumbsUp"
  >
    <Icon
      as="i"
      name="thumbs up"
    />
    <span>
      2
    </span>
  </div>
  <div
    className="thumbsDown"
  >
    <Icon
      as="i"
      name="thumbs down"
    />
  </div>
</div>
`;
